src/Symfony/Component/Lock/StoreInterface.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock;
+
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Exception\NotSupportedException;
+
+/**
+ * StoreInterface defines an interface to manipulate a lock store.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+interface StoreInterface
+{
+    /**
+     * Stores the resource if it's not locked by someone else.
+     *
+     * @param Key $key key to lock
+     *
+     * @throws LockConflictedException
+     */
+    public function save(Key $key);
+
+    /**
+     * Wait a key becomes free, then stores the resource.
+     *
+     * If the store does not supports this feature it should thrown an NotSupportedException.
+     *
+     * @param Key $key key to lock
+     *
+     * @throws LockConflictedException
+     * @throws NotSupportedException
+     */
+    public function waitAndSave(Key $key);
+
+    /**
+     * Extends the ttl of a resource.
+     *
+     * If the store does not supports this feature it should thrown an NotSupportedException.
```

Comment:
If the store does not __support__ this feature it should __throw__ __a__ NotSupportedException.